extensions/amazon-lambda/deployment/src/main/java/io/quarkus/amazon/lambda/deployment/AmazonLambdaProcessor.java

```diff
@@ -92,14 +93,23 @@ AdditionalApplicationArchiveMarkerBuildItem marker() {
 
         for (ClassInfo info : allKnownImplementors) {
             final DotName name = info.name();
-            builder.addBeanClass(name.toString());
+            final String lambda = name.toString();
+
+            // check class is not abstract
+            try {
+                Class<?> lambdaClass = Class.forName(lambda);
+                if (Modifier.isAbstract(lambdaClass.getModifiers())) {
```

Comment:
Hello @oztimpower, instead of loading the class in the processor, this can use the Jandex `ClassInfo` itself to figure out if the class is abstract. Something like:
```
final boolean isAbstract = (info.flags() & java.lang.reflect.Modifier.ABSTRACT) != 0;
```

Reply:
Also `Modifier.isAbstract(info.flasg())` should work